server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -254,19 +274,17 @@ insertBatchResults accumResp Batch{..} remoteResp =
                 -- do we need to assert something about the Right.snd of the result here?
                 -- the remainingRemotes tail (xs) is tossed away (I think?) in caller
                 Seq.Empty ->
-                  case cardinality of
-                    Many -> foldHasuraRowObjs $
-                      \(hasuraRowsSoFar, remainingRemotes) accumData ->
-                         case remainingRemotes of
-                           [] ->
-                             Left $
-                               "no remote objects left for joining at " <> showHashO accumData
-                           (remoteDataVal:rest) -> do
-                             spliced <- spliceRemote remoteDataVal accumData
-                             pure
-                               (OJ.Object spliced : hasuraRowsSoFar , rest)
-                    One ->
-                      Left "Cardinality mismatch: found array in hasura value, but expected object"
+                  -- Since we've reached the end of the path in an Array, we
```

Comment:
Should we make `inValue` similar to `inHashMap` i.e. the first pattern match is on `RelFieldPath` rather than `currentValue`?

Reply:
hm not sure there's a way to do that

Reply:
haha, I think it's like this for a reason then :)